### PR TITLE
Add toBeRequired() method

### PR DESCRIPTION
**Pull request by hiwelo**
**What**:
This pull request is adding `.toBeRequired()` ~and `.toBeOptional()` methods~ allowing folks to test if a form element is ~optional or~ required.

These methods are testing if the form element is having either `aria-required` or `required` attributes.
In the case of a conflict between these two attributes, the value of `required` is having precedence.

**Why**:

This change is following the discussion within the issue #108.

**How**:

I added a new file in `src/` with the different tests inspired from `to-be-disabled.js`.
I added the script to `src/index.js`.
I added a test in `src/__tests__/`.

**Checklist**:

* [x] Documentation
* [x] Tests
* [x] Updated Type Definitions
* [x] Ready to be merged 
* [x] Added myself to contributors table

### CODE REVIEW

File: src/to-be-required.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  return element.tagName && element.tagName.toLowerCase()
+}
+
+function isElementRequired(element) {
+  return FORM_TAGS.includes(getTag(element)) && element.hasAttribute('required')
+}
+
+function isElementRequiredByARIA(element) {
+  return (
+    FORM_TAGS.includes(getTag(element)) &&
```
Comment (gnapse) [inline src/to-be-required.js]: On the other hand I suspect that `aria-selected` is not as restricted, and may be usable even in `div` elements with certain `role`s ([ref](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-required_attribute#Used_in_ARIA_roles)). Though again, I'm not suggesting we go as far as ensuring that if present the element has one of these roles (unless you think is easily doable and makes sense). But at least we should probably not enforce that the element tag is one of these.

File: src/to-be-required.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import {matcherHint, printReceived} from 'jest-matcher-utils'
+import {checkHtmlElement} from './utils'
+
+// form elements that support 'required'
```
[the remark this answers is not in the document]
Comment (hiwelo) [reply src/to-be-required.js]: Yes, thanks for spotting. I will update and I will also have a check to see if the field is read-only. And if it's the case, we are not checking either.

File: src/to-be-required.js (hunk 1 above)
Comment (hiwelo) [reply src/to-be-required.js]: We could enforce that the attribute is present on either:

* a block having `role="combobox"`, `role="gridcell"`, `role="radiogroup"`, `role="spinbutton"` or `role="tree"`
* a block having or a form element corresponding to the `role="listbox"` (so any `<selectbox>`)
* an block having or a form element corresponding to the `role="textbox"` (so any `<input>` and `<textarea>`)

**Comment by hiwelo** [comment]
Also, I wonder if I will not remove `.toBeOptional`. It is just a reverse check but if with the full ARIA implementation I am not sure it makes still sense. Because the lack of `required` or `aria-required="true"` does not mean it is optional. An form requirement enforcement can still occur server-side. What are your thoughts on this @gnapse?

**Comment by hiwelo** [comment]
@gnapse New commits updating the PR with the requested changes.
I also added the different scenarios in the tests and the documentation.

Also, I checked the code coverage & added the type definition.

**Comment by hiwelo** [comment]
@gnapse I just pushed the requested changes. Basically, I removed `.toBeOptional` from:

* the initial script file
* the README.md file
* the type declaration
* the jest tests

Please let me know if something else is needed :+1:

**Comment by hiwelo** [comment]
I just tried to force-push an amended commit to trigger the Travis checks but looks like it is just not working 🙂

**Pull request merged**